### PR TITLE
Job Contract Changes

### PR DESCRIPTION
**Pull request by ericxtang**
Remove `EndJob`

Add a new param (EndBlock) to `Job()`

Change `GetBroadcasterDeposit` to match the new interface

### CODE REVIEW

File: eth/client.go [hunk 1]
```diff
@@ -409,7 +412,7 @@ func (c *Client) WithdrawDeposit() (<-chan types.Receipt, <-chan error) {
 
 func (c *Client) Job(streamId string, transcodingOptions string, maxPricePerSegment *big.Int) (<-chan types.Receipt, <-chan error) {
 	return c.WaitForReceipt(func() (*types.Transaction, error) {
-		if tx, err := c.jobsManagerSession.Job(streamId, transcodingOptions, maxPricePerSegment); err != nil {
+		if tx, err := c.jobsManagerSession.Job(streamId, transcodingOptions, maxPricePerSegment, big.NewInt(0)); err != nil {
```
Comment (yondonfu) [inline eth/client.go]: The endBlock parameter should be non-zero and set to some default value like currentBlock + N and N could be roundLength. We could also let the user specify a custom endBlock, but we should set a default at the very least

**Review by yondonfu — CHANGES_REQUESTED** [review]
Left a comment about setting a non-zero default value for the endBlock when creating a job

**Comment by ericxtang** [comment]
@yondonfu Added a default length of 96 blocks (24 hrs) for now, we can add a new broadcaster option later.

**Review by yondonfu — APPROVED** [review]

**Comment by yondonfu** [comment]
🚢

File: core/livepeernode.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ var EthRpcTimeout = 5 * time.Second
 var EthEventTimeout = 5 * time.Second
 var EthMinedTxTimeout = 60 * time.Second
 var DefaultMasterPlaylistWaitTime = 60 * time.Second
+var DefaultJobLength = int64(96)
```
Comment (dob) [inline core/livepeernode.go:40]: I know this is easily changed, but does this mean that the default max job length is 96 blocks (24 minutes)?

Comment (ericxtang) [reply core/livepeernode.go:40]: Oh yeah... Should be `86400/15=5760`.  Good catch.

**Pull request merged**